[docs] Add brownfield isolated approach guide (#43022)
expo/expo: docs/pages/brownfield/integrated-approach.mdx

This guide will walk you through the steps to add a React Native view into an existing native app. The approach covered here is what we call the "integrated" approach, because React Native and Expo are integrated in the same way that you would any other library.

> Another popular technique is what we call the "isolated" approach, where your Expo app is packaged as a library and treated as a black box by the main existing application. See the [isolated approach guide](/brownfield/isolated-approach/) for details. Now, back to using the "integrated" approach in your existing native app.

Reviewer: ```suggestion
> **info** Another popular technique is what we call the "isolated" approach, where your Expo app is packaged as a library and treated as a black box by the main existing application. See the [isolated approach guide](/brownfield/isolated-approach/) for details.
```

Let's turn this into a callout and and remove the last sentence. This way the visibility of the isolated approach from this doc can be improved.
